### PR TITLE
update(operator): update files for the new API of k8s v1.22 (#252)

### PR DESCRIPTION
**Pull request by dogonthehorizon**
* Update deploy/crds files for the new API of k8s v1.22

* Update dependencies

* Fix conflict with updated dependencies

* Update pkg files and dockerfile.compile for the new API of k8s v1.22

* Update dependencies and CRDs

* Edit logger in test files

* fix unit test files

* fix(CRDs): Change openapi command to use controller-gen and update go version

* fix(tests): Update unit testing

* fix(ubi): Remove Ubi

* fix(ubi): Double quote error

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,68 +2,131 @@ module github.com/armory/spinnaker-operator
...
 
 // Remove when controller-tools v0.2.2 is released
-// Required for the bugfix https://github.com/kubernetes-sigs/controller-tools/pull/322
 replace sigs.k8s.io/controller-tools => sigs.k8s.io/controller-tools v0.2.2-0.20190919011008-6ed4ff330711
 
-replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v0.10.1-0.20191004014855-dc713e4d7890
+// replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v0.18.0
+replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v0.19.4
 
 replace github.com/operator-framework/operator-registry => github.com/operator-framework/operator-registry v1.2.0
```
Comment (dogonthehorizon) [inline go.mod:130]: We should be able to remove all of this

File: pkg/apis/spinnaker/interfaces/interfaces.go [hunk 2]
```diff
@@ -59,12 +64,22 @@ type SpinnakerService interface {
...
 	GetItems() []SpinnakerService
 	DeepCopySpinnakerServiceList() SpinnakerServiceList
+	GetContinue() string
+	GetRemainingItemCount() *int64
+	GetResourceVersion() string
+	GetSelfLink() string
+	SetContinue(c string)
+	SetRemainingItemCount(c *int64)
+	SetResourceVersion(c string)
+	SetSelfLink(c string)
```
Comment (dogonthehorizon) [inline pkg/apis/spinnaker/interfaces/interfaces.go:79]: Do we still need these?

File: pkg/controller/accountvalidating/accountvalidating_controller.go [hunk 3]
```diff
@@ -3,21 +3,22 @@ package accountvalidating
...
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
-// +kubebuilder:webhook:path=/validate-v1-spinnakerservice,mutating=false,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=vpod.kb.io
+// +kubebuilder:webhook:path=/validate-v1-spinnakerservice,mutating=false,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=vpod.kb.io,admissionReviewVersions=v1,sideEffects=none
```
Comment (dogonthehorizon) [inline pkg/controller/accountvalidating/accountvalidating_controller.go:21]: This is what was generating the webhook manifest, we should remove this

File: pkg/apis/spinnaker/interfaces/interfaces.go (hunk 2 above)
Comment (ichi0915) [reply pkg/apis/spinnaker/interfaces/interfaces.go:78]: We need it for the library `sigs.k8s.io/controller-runtime v0.11.2`

**Review by ichi0915 — APPROVED** [review]

**Comment by dogonthehorizon** [comment]
Previous container build/test cycle passed, don't want to wait for this the latest commit to finish since it only introduces Mergify changes. These config changes will not take effect until they're merged to master, so going ahead w/ this merge.

**Pull request merged**